// comfydeploy/models/errors/getdeployment.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetDeploymentResponseBody$ = exports.GetDeploymentResponseBody$outboundSchema = exports.GetDeploymentResponseBody$inboundSchema = exports.GetDeploymentResponseBody = void 0;
const z = __importStar(require("zod"));
/**
 * error
 */
class GetDeploymentResponseBody extends Error {
    constructor(err) {
        const message = "message" in err && typeof err.message === "string"
            ? err.message
            : `API error occurred: ${JSON.stringify(err)}`;
        super(message);
        this.data$ = err;
        this.error = err.error;
        this.name = "GetDeploymentResponseBody";
    }
}
exports.GetDeploymentResponseBody = GetDeploymentResponseBody;
/** @internal */
exports.GetDeploymentResponseBody$inboundSchema = z.object({
    error: z.string(),
})
    .transform((v) => {
    return new GetDeploymentResponseBody(v);
});
/** @internal */
exports.GetDeploymentResponseBody$outboundSchema = z.instanceof(GetDeploymentResponseBody)
    .transform(v => v.data$)
    .pipe(z.object({
    error: z.string(),
}));
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetDeploymentResponseBody$;
(function (GetDeploymentResponseBody$) {
    /** @deprecated use `GetDeploymentResponseBody$inboundSchema` instead. */
    GetDeploymentResponseBody$.inboundSchema = exports.GetDeploymentResponseBody$inboundSchema;
    /** @deprecated use `GetDeploymentResponseBody$outboundSchema` instead. */
    GetDeploymentResponseBody$.outboundSchema = exports.GetDeploymentResponseBody$outboundSchema;
})(GetDeploymentResponseBody$ || (exports.GetDeploymentResponseBody$ = GetDeploymentResponseBody$ = {}));
