// comfydeploy/models/errors/getdeploymentidinputs.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetDeploymentIdInputsResponseBody$ = exports.GetDeploymentIdInputsResponseBody$outboundSchema = exports.GetDeploymentIdInputsResponseBody$inboundSchema = exports.GetDeploymentIdInputsResponseBody = void 0;
const z = __importStar(require("zod"));
/**
 * Error when uploading the workflow
 */
class GetDeploymentIdInputsResponseBody extends Error {
    constructor(err) {
        const message = "message" in err && typeof err.message === "string"
            ? err.message
            : `API error occurred: ${JSON.stringify(err)}`;
        super(message);
        this.data$ = err;
        this.error = err.error;
        this.name = "GetDeploymentIdInputsResponseBody";
    }
}
exports.GetDeploymentIdInputsResponseBody = GetDeploymentIdInputsResponseBody;
/** @internal */
exports.GetDeploymentIdInputsResponseBody$inboundSchema = z.object({
    error: z.string(),
})
    .transform((v) => {
    return new GetDeploymentIdInputsResponseBody(v);
});
/** @internal */
exports.GetDeploymentIdInputsResponseBody$outboundSchema = z.instanceof(GetDeploymentIdInputsResponseBody)
    .transform(v => v.data$)
    .pipe(z.object({
    error: z.string(),
}));
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetDeploymentIdInputsResponseBody$;
(function (GetDeploymentIdInputsResponseBody$) {
    /** @deprecated use `GetDeploymentIdInputsResponseBody$inboundSchema` instead. */
    GetDeploymentIdInputsResponseBody$.inboundSchema = exports.GetDeploymentIdInputsResponseBody$inboundSchema;
    /** @deprecated use `GetDeploymentIdInputsResponseBody$outboundSchema` instead. */
    GetDeploymentIdInputsResponseBody$.outboundSchema = exports.GetDeploymentIdInputsResponseBody$outboundSchema;
})(GetDeploymentIdInputsResponseBody$ || (exports.GetDeploymentIdInputsResponseBody$ = GetDeploymentIdInputsResponseBody$ = {}));
