// comfydeploy/models/components/apiroutestypesworkflowrunoutputmodel2.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ApiRoutesTypesWorkflowRunOutputModel2$ = exports.ApiRoutesTypesWorkflowRunOutputModel2$outboundSchema = exports.ApiRoutesTypesWorkflowRunOutputModel2$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const mediaitem_js_1 = require("./mediaitem.js");
/** @internal */
exports.ApiRoutesTypesWorkflowRunOutputModel2$inboundSchema = z.object({
    id: z.string(),
    run_id: z.string(),
    data: z.record(z.array(mediaitem_js_1.MediaItem$inboundSchema)),
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    updated_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    type: z.nullable(z.string()).optional(),
    node_id: z.nullable(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "run_id": "runId",
        "created_at": "createdAt",
        "updated_at": "updatedAt",
        "node_id": "nodeId",
    });
});
/** @internal */
exports.ApiRoutesTypesWorkflowRunOutputModel2$outboundSchema = z.object({
    id: z.string(),
    runId: z.string(),
    data: z.record(z.array(mediaitem_js_1.MediaItem$outboundSchema)),
    createdAt: z.date().transform(v => v.toISOString()),
    updatedAt: z.date().transform(v => v.toISOString()),
    type: z.nullable(z.string()).optional(),
    nodeId: z.nullable(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        runId: "run_id",
        createdAt: "created_at",
        updatedAt: "updated_at",
        nodeId: "node_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ApiRoutesTypesWorkflowRunOutputModel2$;
(function (ApiRoutesTypesWorkflowRunOutputModel2$) {
    /** @deprecated use `ApiRoutesTypesWorkflowRunOutputModel2$inboundSchema` instead. */
    ApiRoutesTypesWorkflowRunOutputModel2$.inboundSchema = exports.ApiRoutesTypesWorkflowRunOutputModel2$inboundSchema;
    /** @deprecated use `ApiRoutesTypesWorkflowRunOutputModel2$outboundSchema` instead. */
    ApiRoutesTypesWorkflowRunOutputModel2$.outboundSchema = exports.ApiRoutesTypesWorkflowRunOutputModel2$outboundSchema;
})(ApiRoutesTypesWorkflowRunOutputModel2$ || (exports.ApiRoutesTypesWorkflowRunOutputModel2$ = ApiRoutesTypesWorkflowRunOutputModel2$ = {}));
