// comfydeploy/models/components/apiroutestypesworkflowrunoutputmodel1.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ApiRoutesTypesWorkflowRunOutputModel1$ = exports.ApiRoutesTypesWorkflowRunOutputModel1$outboundSchema = exports.ApiRoutesTypesWorkflowRunOutputModel1$inboundSchema = exports.NodeMeta$ = exports.NodeMeta$outboundSchema = exports.NodeMeta$inboundSchema = exports.Data$ = exports.Data$outboundSchema = exports.Data$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
/** @internal */
exports.Data$inboundSchema = z
    .object({});
/** @internal */
exports.Data$outboundSchema = z.object({});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Data$;
(function (Data$) {
    /** @deprecated use `Data$inboundSchema` instead. */
    Data$.inboundSchema = exports.Data$inboundSchema;
    /** @deprecated use `Data$outboundSchema` instead. */
    Data$.outboundSchema = exports.Data$outboundSchema;
})(Data$ || (exports.Data$ = Data$ = {}));
/** @internal */
exports.NodeMeta$inboundSchema = z.object({});
/** @internal */
exports.NodeMeta$outboundSchema = z.object({});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var NodeMeta$;
(function (NodeMeta$) {
    /** @deprecated use `NodeMeta$inboundSchema` instead. */
    NodeMeta$.inboundSchema = exports.NodeMeta$inboundSchema;
    /** @deprecated use `NodeMeta$outboundSchema` instead. */
    NodeMeta$.outboundSchema = exports.NodeMeta$outboundSchema;
})(NodeMeta$ || (exports.NodeMeta$ = NodeMeta$ = {}));
/** @internal */
exports.ApiRoutesTypesWorkflowRunOutputModel1$inboundSchema = z.object({
    id: z.string(),
    run_id: z.string(),
    data: z.nullable(z.lazy(() => exports.Data$inboundSchema)),
    node_meta: z.nullable(z.lazy(() => exports.NodeMeta$inboundSchema)),
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    updated_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "run_id": "runId",
        "node_meta": "nodeMeta",
        "created_at": "createdAt",
        "updated_at": "updatedAt",
    });
});
/** @internal */
exports.ApiRoutesTypesWorkflowRunOutputModel1$outboundSchema = z.object({
    id: z.string(),
    runId: z.string(),
    data: z.nullable(z.lazy(() => exports.Data$outboundSchema)),
    nodeMeta: z.nullable(z.lazy(() => exports.NodeMeta$outboundSchema)),
    createdAt: z.date().transform(v => v.toISOString()),
    updatedAt: z.date().transform(v => v.toISOString()),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        runId: "run_id",
        nodeMeta: "node_meta",
        createdAt: "created_at",
        updatedAt: "updated_at",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ApiRoutesTypesWorkflowRunOutputModel1$;
(function (ApiRoutesTypesWorkflowRunOutputModel1$) {
    /** @deprecated use `ApiRoutesTypesWorkflowRunOutputModel1$inboundSchema` instead. */
    ApiRoutesTypesWorkflowRunOutputModel1$.inboundSchema = exports.ApiRoutesTypesWorkflowRunOutputModel1$inboundSchema;
    /** @deprecated use `ApiRoutesTypesWorkflowRunOutputModel1$outboundSchema` instead. */
    ApiRoutesTypesWorkflowRunOutputModel1$.outboundSchema = exports.ApiRoutesTypesWorkflowRunOutputModel1$outboundSchema;
})(ApiRoutesTypesWorkflowRunOutputModel1$ || (exports.ApiRoutesTypesWorkflowRunOutputModel1$ = ApiRoutesTypesWorkflowRunOutputModel1$ = {}));
