// comfydeploy/models/components/createdynamicsessionbody.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateDynamicSessionBody$ = exports.CreateDynamicSessionBody$outboundSchema = exports.CreateDynamicSessionBody$inboundSchema = exports.Dependencies$ = exports.Dependencies$outboundSchema = exports.Dependencies$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const depsbody_js_1 = require("./depsbody.js");
const machinegpu_js_1 = require("./machinegpu.js");
/** @internal */
exports.Dependencies$inboundSchema = z.union([depsbody_js_1.DepsBody$inboundSchema, z.array(z.string())]);
/** @internal */
exports.Dependencies$outboundSchema = z.union([depsbody_js_1.DepsBody$outboundSchema, z.array(z.string())]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Dependencies$;
(function (Dependencies$) {
    /** @deprecated use `Dependencies$inboundSchema` instead. */
    Dependencies$.inboundSchema = exports.Dependencies$inboundSchema;
    /** @deprecated use `Dependencies$outboundSchema` instead. */
    Dependencies$.outboundSchema = exports.Dependencies$outboundSchema;
})(Dependencies$ || (exports.Dependencies$ = Dependencies$ = {}));
/** @internal */
exports.CreateDynamicSessionBody$inboundSchema = z.object({
    machine_id: z.string(),
    gpu: machinegpu_js_1.MachineGPU$inboundSchema.optional(),
    timeout: z.nullable(z.number().int()).optional(),
    dependencies: z.nullable(z.union([depsbody_js_1.DepsBody$inboundSchema, z.array(z.string())])).optional(),
    wait_for_server: z.boolean().default(false),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "machine_id": "machineId",
        "wait_for_server": "waitForServer",
    });
});
/** @internal */
exports.CreateDynamicSessionBody$outboundSchema = z.object({
    machineId: z.string(),
    gpu: machinegpu_js_1.MachineGPU$outboundSchema.optional(),
    timeout: z.nullable(z.number().int()).optional(),
    dependencies: z.nullable(z.union([depsbody_js_1.DepsBody$outboundSchema, z.array(z.string())])).optional(),
    waitForServer: z.boolean().default(false),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        machineId: "machine_id",
        waitForServer: "wait_for_server",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CreateDynamicSessionBody$;
(function (CreateDynamicSessionBody$) {
    /** @deprecated use `CreateDynamicSessionBody$inboundSchema` instead. */
    CreateDynamicSessionBody$.inboundSchema = exports.CreateDynamicSessionBody$inboundSchema;
    /** @deprecated use `CreateDynamicSessionBody$outboundSchema` instead. */
    CreateDynamicSessionBody$.outboundSchema = exports.CreateDynamicSessionBody$outboundSchema;
})(CreateDynamicSessionBody$ || (exports.CreateDynamicSessionBody$ = CreateDynamicSessionBody$ = {}));
