// comfydeploy/models/components/createsessionbody.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateSessionBody$ = exports.CreateSessionBody$outboundSchema = exports.CreateSessionBody$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const machinegpu_js_1 = require("./machinegpu.js");
/** @internal */
exports.CreateSessionBody$inboundSchema = z.object({
    machine_id: z.string(),
    gpu: z.nullable(machinegpu_js_1.MachineGPU$inboundSchema).optional(),
    timeout: z.nullable(z.number().int()).optional(),
    wait_for_server: z.boolean().default(false),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "machine_id": "machineId",
        "wait_for_server": "waitForServer",
    });
});
/** @internal */
exports.CreateSessionBody$outboundSchema = z.object({
    machineId: z.string(),
    gpu: z.nullable(machinegpu_js_1.MachineGPU$outboundSchema).optional(),
    timeout: z.nullable(z.number().int()).optional(),
    waitForServer: z.boolean().default(false),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        machineId: "machine_id",
        waitForServer: "wait_for_server",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CreateSessionBody$;
(function (CreateSessionBody$) {
    /** @deprecated use `CreateSessionBody$inboundSchema` instead. */
    CreateSessionBody$.inboundSchema = exports.CreateSessionBody$inboundSchema;
    /** @deprecated use `CreateSessionBody$outboundSchema` instead. */
    CreateSessionBody$.outboundSchema = exports.CreateSessionBody$outboundSchema;
})(CreateSessionBody$ || (exports.CreateSessionBody$ = CreateSessionBody$ = {}));
