// comfydeploy/models/components/customnode.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomNode$ = exports.CustomNode$outboundSchema = exports.CustomNode$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
/** @internal */
exports.CustomNode$inboundSchema = z.object({
    pip: z.nullable(z.array(z.string())).optional(),
    url: z.string(),
    hash: z.nullable(z.string()).optional(),
    install_type: z.string(),
    files: z.nullable(z.array(z.string())).optional(),
    name: z.nullable(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "install_type": "installType",
    });
});
/** @internal */
exports.CustomNode$outboundSchema = z.object({
    pip: z.nullable(z.array(z.string())).optional(),
    url: z.string(),
    hash: z.nullable(z.string()).optional(),
    installType: z.string(),
    files: z.nullable(z.array(z.string())).optional(),
    name: z.nullable(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        installType: "install_type",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomNode$;
(function (CustomNode$) {
    /** @deprecated use `CustomNode$inboundSchema` instead. */
    CustomNode$.inboundSchema = exports.CustomNode$inboundSchema;
    /** @deprecated use `CustomNode$outboundSchema` instead. */
    CustomNode$.outboundSchema = exports.CustomNode$outboundSchema;
})(CustomNode$ || (exports.CustomNode$ = CustomNode$ = {}));
