// comfydeploy/models/components/dependencygraph.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DependencyGraph$ = exports.DependencyGraph$outboundSchema = exports.DependencyGraph$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const customnode_js_1 = require("./customnode.js");
/** @internal */
exports.DependencyGraph$inboundSchema = z.object({
    comfyui: z.string(),
    models: z.record(z.string()),
    missing_nodes: z.array(z.string()),
    custom_nodes: z.record(customnode_js_1.CustomNode$inboundSchema),
    files: z.record(z.string()),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "missing_nodes": "missingNodes",
        "custom_nodes": "customNodes",
    });
});
/** @internal */
exports.DependencyGraph$outboundSchema = z.object({
    comfyui: z.string(),
    models: z.record(z.string()),
    missingNodes: z.array(z.string()),
    customNodes: z.record(customnode_js_1.CustomNode$outboundSchema),
    files: z.record(z.string()),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        missingNodes: "missing_nodes",
        customNodes: "custom_nodes",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DependencyGraph$;
(function (DependencyGraph$) {
    /** @deprecated use `DependencyGraph$inboundSchema` instead. */
    DependencyGraph$.inboundSchema = exports.DependencyGraph$inboundSchema;
    /** @deprecated use `DependencyGraph$outboundSchema` instead. */
    DependencyGraph$.outboundSchema = exports.DependencyGraph$outboundSchema;
})(DependencyGraph$ || (exports.DependencyGraph$ = DependencyGraph$ = {}));
