// comfydeploy/models/components/depsbody.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DepsBody$ = exports.DepsBody$outboundSchema = exports.DepsBody$inboundSchema = exports.Snapshot$ = exports.Snapshot$outboundSchema = exports.Snapshot$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const dependencygraph_js_1 = require("./dependencygraph.js");
/** @internal */
exports.Snapshot$inboundSchema = z.object({});
/** @internal */
exports.Snapshot$outboundSchema = z.object({});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Snapshot$;
(function (Snapshot$) {
    /** @deprecated use `Snapshot$inboundSchema` instead. */
    Snapshot$.inboundSchema = exports.Snapshot$inboundSchema;
    /** @deprecated use `Snapshot$outboundSchema` instead. */
    Snapshot$.outboundSchema = exports.Snapshot$outboundSchema;
})(Snapshot$ || (exports.Snapshot$ = Snapshot$ = {}));
/** @internal */
exports.DepsBody$inboundSchema = z.object({
    docker_command_steps: z.any().optional(),
    dependencies: z.nullable(dependencygraph_js_1.DependencyGraph$inboundSchema).optional(),
    snapshot: z.nullable(z.lazy(() => exports.Snapshot$inboundSchema)).optional(),
    comfyui_version: z.string().default("8ce2a1052ca03183768da0aaa483024e58b8008c"),
    extra_docker_commands: z.nullable(z.array(z.record(z.string()))).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "docker_command_steps": "dockerCommandSteps",
        "comfyui_version": "comfyuiVersion",
        "extra_docker_commands": "extraDockerCommands",
    });
});
/** @internal */
exports.DepsBody$outboundSchema = z.object({
    dockerCommandSteps: z.any().optional(),
    dependencies: z.nullable(dependencygraph_js_1.DependencyGraph$outboundSchema).optional(),
    snapshot: z.nullable(z.lazy(() => exports.Snapshot$outboundSchema)).optional(),
    comfyuiVersion: z.string().default("8ce2a1052ca03183768da0aaa483024e58b8008c"),
    extraDockerCommands: z.nullable(z.array(z.record(z.string()))).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        dockerCommandSteps: "docker_command_steps",
        comfyuiVersion: "comfyui_version",
        extraDockerCommands: "extra_docker_commands",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DepsBody$;
(function (DepsBody$) {
    /** @deprecated use `DepsBody$inboundSchema` instead. */
    DepsBody$.inboundSchema = exports.DepsBody$inboundSchema;
    /** @deprecated use `DepsBody$outboundSchema` instead. */
    DepsBody$.outboundSchema = exports.DepsBody$outboundSchema;
})(DepsBody$ || (exports.DepsBody$ = DepsBody$ = {}));
