// comfydeploy/models/components/fileuploadresponse.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FileUploadResponse$ = exports.FileUploadResponse$outboundSchema = exports.FileUploadResponse$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
/** @internal */
exports.FileUploadResponse$inboundSchema = z.object({
    message: z.string(),
    file_id: z.string(),
    file_name: z.string(),
    file_url: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "file_id": "fileId",
        "file_name": "fileName",
        "file_url": "fileUrl",
    });
});
/** @internal */
exports.FileUploadResponse$outboundSchema = z.object({
    message: z.string(),
    fileId: z.string(),
    fileName: z.string(),
    fileUrl: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        fileId: "file_id",
        fileName: "file_name",
        fileUrl: "file_url",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var FileUploadResponse$;
(function (FileUploadResponse$) {
    /** @deprecated use `FileUploadResponse$inboundSchema` instead. */
    FileUploadResponse$.inboundSchema = exports.FileUploadResponse$inboundSchema;
    /** @deprecated use `FileUploadResponse$outboundSchema` instead. */
    FileUploadResponse$.outboundSchema = exports.FileUploadResponse$outboundSchema;
})(FileUploadResponse$ || (exports.FileUploadResponse$ = FileUploadResponse$ = {}));
