// comfydeploy/models/components/logupdateevent.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.LogUpdateEvent$ = exports.LogUpdateEvent$outboundSchema = exports.LogUpdateEvent$inboundSchema = exports.Event$ = exports.Event$outboundSchema = exports.Event$inboundSchema = exports.Event = void 0;
const z = __importStar(require("zod"));
const logdatacontent_js_1 = require("./logdatacontent.js");
exports.Event = {
    LogUpdate: "log_update",
};
/** @internal */
exports.Event$inboundSchema = z.nativeEnum(exports.Event);
/** @internal */
exports.Event$outboundSchema = exports.Event$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Event$;
(function (Event$) {
    /** @deprecated use `Event$inboundSchema` instead. */
    Event$.inboundSchema = exports.Event$inboundSchema;
    /** @deprecated use `Event$outboundSchema` instead. */
    Event$.outboundSchema = exports.Event$outboundSchema;
})(Event$ || (exports.Event$ = Event$ = {}));
/** @internal */
exports.LogUpdateEvent$inboundSchema = z.object({
    event: z.literal("log_update").default("log_update"),
    data: z.string().transform((v, ctx) => {
        try {
            return JSON.parse(v);
        }
        catch (err) {
            ctx.addIssue({
                code: z.ZodIssueCode.custom,
                message: `malformed json: ${err}`,
            });
            return z.NEVER;
        }
    }).pipe(logdatacontent_js_1.LogDataContent$inboundSchema),
});
/** @internal */
exports.LogUpdateEvent$outboundSchema = z.object({
    event: z.literal("log_update").default("log_update"),
    data: logdatacontent_js_1.LogDataContent$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var LogUpdateEvent$;
(function (LogUpdateEvent$) {
    /** @deprecated use `LogUpdateEvent$inboundSchema` instead. */
    LogUpdateEvent$.inboundSchema = exports.LogUpdateEvent$inboundSchema;
    /** @deprecated use `LogUpdateEvent$outboundSchema` instead. */
    LogUpdateEvent$.outboundSchema = exports.LogUpdateEvent$outboundSchema;
})(LogUpdateEvent$ || (exports.LogUpdateEvent$ = LogUpdateEvent$ = {}));
