// comfydeploy/models/components/mediaoutput.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MediaOutput$ = exports.MediaOutput$outboundSchema = exports.MediaOutput$inboundSchema = void 0;
const z = __importStar(require("zod"));
const mediaitem_js_1 = require("./mediaitem.js");
/** @internal */
exports.MediaOutput$inboundSchema = z.object({
    images: z.nullable(z.array(mediaitem_js_1.MediaItem$inboundSchema)).optional(),
    gifs: z.nullable(z.array(mediaitem_js_1.MediaItem$inboundSchema)).optional(),
    videos: z.nullable(z.array(mediaitem_js_1.MediaItem$inboundSchema)).optional(),
    other: z.nullable(z.array(mediaitem_js_1.MediaItem$inboundSchema)).optional(),
});
/** @internal */
exports.MediaOutput$outboundSchema = z.object({
    images: z.nullable(z.array(mediaitem_js_1.MediaItem$outboundSchema)).optional(),
    gifs: z.nullable(z.array(mediaitem_js_1.MediaItem$outboundSchema)).optional(),
    videos: z.nullable(z.array(mediaitem_js_1.MediaItem$outboundSchema)).optional(),
    other: z.nullable(z.array(mediaitem_js_1.MediaItem$outboundSchema)).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var MediaOutput$;
(function (MediaOutput$) {
    /** @deprecated use `MediaOutput$inboundSchema` instead. */
    MediaOutput$.inboundSchema = exports.MediaOutput$inboundSchema;
    /** @deprecated use `MediaOutput$outboundSchema` instead. */
    MediaOutput$.outboundSchema = exports.MediaOutput$outboundSchema;
})(MediaOutput$ || (exports.MediaOutput$ = MediaOutput$ = {}));
