// comfydeploy/models/components/mediaitem.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MediaItem$ = exports.MediaItem$outboundSchema = exports.MediaItem$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
/** @internal */
exports.MediaItem$inboundSchema = z.object({
    url: z.string(),
    type: z.string(),
    filename: z.string(),
    is_public: z.nullable(z.boolean()).optional(),
    subfolder: z.nullable(z.string()).optional(),
    upload_duration: z.nullable(z.number()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "is_public": "isPublic",
        "upload_duration": "uploadDuration",
    });
});
/** @internal */
exports.MediaItem$outboundSchema = z.object({
    url: z.string(),
    type: z.string(),
    filename: z.string(),
    isPublic: z.nullable(z.boolean()).optional(),
    subfolder: z.nullable(z.string()).optional(),
    uploadDuration: z.nullable(z.number()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        isPublic: "is_public",
        uploadDuration: "upload_duration",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var MediaItem$;
(function (MediaItem$) {
    /** @deprecated use `MediaItem$inboundSchema` instead. */
    MediaItem$.inboundSchema = exports.MediaItem$inboundSchema;
    /** @deprecated use `MediaItem$outboundSchema` instead. */
    MediaItem$.outboundSchema = exports.MediaItem$outboundSchema;
})(MediaItem$ || (exports.MediaItem$ = MediaItem$ = {}));
