// comfydeploy/models/components/session.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Session$ = exports.Session$outboundSchema = exports.Session$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
/** @internal */
exports.Session$inboundSchema = z.object({
    session_id: z.string(),
    gpu_event_id: z.string(),
    url: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "session_id": "sessionId",
        "gpu_event_id": "gpuEventId",
    });
});
/** @internal */
exports.Session$outboundSchema = z.object({
    sessionId: z.string(),
    gpuEventId: z.string(),
    url: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        sessionId: "session_id",
        gpuEventId: "gpu_event_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Session$;
(function (Session$) {
    /** @deprecated use `Session$inboundSchema` instead. */
    Session$.inboundSchema = exports.Session$inboundSchema;
    /** @deprecated use `Session$outboundSchema` instead. */
    Session$.outboundSchema = exports.Session$outboundSchema;
})(Session$ || (exports.Session$ = Session$ = {}));
