// comfydeploy/models/components/workflowrunwebhookbody.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkflowRunWebhookBody$ = exports.WorkflowRunWebhookBody$outboundSchema = exports.WorkflowRunWebhookBody$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const workflowrunoutputmodel_js_1 = require("./workflowrunoutputmodel.js");
const workflowrunstatus_js_1 = require("./workflowrunstatus.js");
/** @internal */
exports.WorkflowRunWebhookBody$inboundSchema = z.object({
    run_id: z.string(),
    status: workflowrunstatus_js_1.WorkflowRunStatus$inboundSchema,
    live_status: z.nullable(z.string()),
    progress: z.number().default(0),
    outputs: z.array(workflowrunoutputmodel_js_1.WorkflowRunOutputModel$inboundSchema).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "run_id": "runId",
        "live_status": "liveStatus",
    });
});
/** @internal */
exports.WorkflowRunWebhookBody$outboundSchema = z.object({
    runId: z.string(),
    status: workflowrunstatus_js_1.WorkflowRunStatus$outboundSchema,
    liveStatus: z.nullable(z.string()),
    progress: z.number().default(0),
    outputs: z.array(workflowrunoutputmodel_js_1.WorkflowRunOutputModel$outboundSchema).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        runId: "run_id",
        liveStatus: "live_status",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WorkflowRunWebhookBody$;
(function (WorkflowRunWebhookBody$) {
    /** @deprecated use `WorkflowRunWebhookBody$inboundSchema` instead. */
    WorkflowRunWebhookBody$.inboundSchema = exports.WorkflowRunWebhookBody$inboundSchema;
    /** @deprecated use `WorkflowRunWebhookBody$outboundSchema` instead. */
    WorkflowRunWebhookBody$.outboundSchema = exports.WorkflowRunWebhookBody$outboundSchema;
})(WorkflowRunWebhookBody$ || (exports.WorkflowRunWebhookBody$ = WorkflowRunWebhookBody$ = {}));
