// comfydeploy/models/components/workflowrunstatus.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkflowRunStatus$ = exports.WorkflowRunStatus$outboundSchema = exports.WorkflowRunStatus$inboundSchema = exports.WorkflowRunStatus = void 0;
const z = __importStar(require("zod"));
exports.WorkflowRunStatus = {
    NotStarted: "not-started",
    Running: "running",
    Uploading: "uploading",
    Success: "success",
    Failed: "failed",
    Started: "started",
    Queued: "queued",
    Timeout: "timeout",
    Cancelled: "cancelled",
};
/** @internal */
exports.WorkflowRunStatus$inboundSchema = z.nativeEnum(exports.WorkflowRunStatus);
/** @internal */
exports.WorkflowRunStatus$outboundSchema = exports.WorkflowRunStatus$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WorkflowRunStatus$;
(function (WorkflowRunStatus$) {
    /** @deprecated use `WorkflowRunStatus$inboundSchema` instead. */
    WorkflowRunStatus$.inboundSchema = exports.WorkflowRunStatus$inboundSchema;
    /** @deprecated use `WorkflowRunStatus$outboundSchema` instead. */
    WorkflowRunStatus$.outboundSchema = exports.WorkflowRunStatus$outboundSchema;
})(WorkflowRunStatus$ || (exports.WorkflowRunStatus$ = WorkflowRunStatus$ = {}));
