// comfydeploy/models/components/workflowrunnativeoutputmodel.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkflowRunNativeOutputModel$ = exports.WorkflowRunNativeOutputModel$outboundSchema = exports.WorkflowRunNativeOutputModel$inboundSchema = exports.WorkflowRunNativeOutputModelInputs$ = exports.WorkflowRunNativeOutputModelInputs$outboundSchema = exports.WorkflowRunNativeOutputModelInputs$inboundSchema = exports.WorkflowApiRaw$ = exports.WorkflowApiRaw$outboundSchema = exports.WorkflowApiRaw$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
/** @internal */
exports.WorkflowApiRaw$inboundSchema = z.object({});
/** @internal */
exports.WorkflowApiRaw$outboundSchema = z.object({});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WorkflowApiRaw$;
(function (WorkflowApiRaw$) {
    /** @deprecated use `WorkflowApiRaw$inboundSchema` instead. */
    WorkflowApiRaw$.inboundSchema = exports.WorkflowApiRaw$inboundSchema;
    /** @deprecated use `WorkflowApiRaw$outboundSchema` instead. */
    WorkflowApiRaw$.outboundSchema = exports.WorkflowApiRaw$outboundSchema;
})(WorkflowApiRaw$ || (exports.WorkflowApiRaw$ = WorkflowApiRaw$ = {}));
/** @internal */
exports.WorkflowRunNativeOutputModelInputs$inboundSchema = z.object({});
/** @internal */
exports.WorkflowRunNativeOutputModelInputs$outboundSchema = z.object({});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WorkflowRunNativeOutputModelInputs$;
(function (WorkflowRunNativeOutputModelInputs$) {
    /** @deprecated use `WorkflowRunNativeOutputModelInputs$inboundSchema` instead. */
    WorkflowRunNativeOutputModelInputs$.inboundSchema = exports.WorkflowRunNativeOutputModelInputs$inboundSchema;
    /** @deprecated use `WorkflowRunNativeOutputModelInputs$outboundSchema` instead. */
    WorkflowRunNativeOutputModelInputs$.outboundSchema = exports.WorkflowRunNativeOutputModelInputs$outboundSchema;
})(WorkflowRunNativeOutputModelInputs$ || (exports.WorkflowRunNativeOutputModelInputs$ = WorkflowRunNativeOutputModelInputs$ = {}));
/** @internal */
exports.WorkflowRunNativeOutputModel$inboundSchema = z.object({
    prompt_id: z.string(),
    workflow_api_raw: z.lazy(() => exports.WorkflowApiRaw$inboundSchema),
    inputs: z.nullable(z.lazy(() => exports.WorkflowRunNativeOutputModelInputs$inboundSchema)),
    status_endpoint: z.string(),
    file_upload_endpoint: z.string(),
    cd_token: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "prompt_id": "promptId",
        "workflow_api_raw": "workflowApiRaw",
        "status_endpoint": "statusEndpoint",
        "file_upload_endpoint": "fileUploadEndpoint",
        "cd_token": "cdToken",
    });
});
/** @internal */
exports.WorkflowRunNativeOutputModel$outboundSchema = z.object({
    promptId: z.string(),
    workflowApiRaw: z.lazy(() => exports.WorkflowApiRaw$outboundSchema),
    inputs: z.nullable(z.lazy(() => exports.WorkflowRunNativeOutputModelInputs$outboundSchema)),
    statusEndpoint: z.string(),
    fileUploadEndpoint: z.string(),
    cdToken: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        promptId: "prompt_id",
        workflowApiRaw: "workflow_api_raw",
        statusEndpoint: "status_endpoint",
        fileUploadEndpoint: "file_upload_endpoint",
        cdToken: "cd_token",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WorkflowRunNativeOutputModel$;
(function (WorkflowRunNativeOutputModel$) {
    /** @deprecated use `WorkflowRunNativeOutputModel$inboundSchema` instead. */
    WorkflowRunNativeOutputModel$.inboundSchema = exports.WorkflowRunNativeOutputModel$inboundSchema;
    /** @deprecated use `WorkflowRunNativeOutputModel$outboundSchema` instead. */
    WorkflowRunNativeOutputModel$.outboundSchema = exports.WorkflowRunNativeOutputModel$outboundSchema;
})(WorkflowRunNativeOutputModel$ || (exports.WorkflowRunNativeOutputModel$ = WorkflowRunNativeOutputModel$ = {}));
