// comfydeploy/sdk/machine.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Machine = void 0;
const machineGetV1Machines_js_1 = require("../funcs/machineGetV1Machines.js");
const machineGetV1MachinesMachineId_js_1 = require("../funcs/machineGetV1MachinesMachineId.js");
const machineListEvents_js_1 = require("../funcs/machineListEvents.js");
const machinePostGpuEvent_js_1 = require("../funcs/machinePostGpuEvent.js");
const machinePostV1Machines_js_1 = require("../funcs/machinePostV1Machines.js");
const sdks_js_1 = require("../lib/sdks.js");
const fp_js_1 = require("../types/fp.js");
class Machine extends sdks_js_1.ClientSDK {
    /**
     * Register a machine event
     */
    async postGpuEvent(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, machinePostGpuEvent_js_1.machinePostGpuEvent)(this, request, options));
    }
    /**
     * Get recent gpu events
     */
    async listEvents(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, machineListEvents_js_1.machineListEvents)(this, request, options));
    }
    /**
     * Create a new machine
     *
     * @remarks
     * Create a new machine with optional default setting
     */
    async postV1Machines(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, machinePostV1Machines_js_1.machinePostV1Machines)(this, request, options));
    }
    /**
     * Retrieve all machines for a user
     *
     * @remarks
     * Retrieve details of all machines for the authenticated user, with pagination and optional field selection
     */
    async getV1Machines(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, machineGetV1Machines_js_1.machineGetV1Machines)(this, request, options));
    }
    /**
     * Retrieve a specific machine by ID
     *
     * @remarks
     * Retrieve details of a specific machine by its ID, with optional workspace details
     */
    async getV1MachinesMachineId(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, machineGetV1MachinesMachineId_js_1.machineGetV1MachinesMachineId)(this, request, options));
    }
}
exports.Machine = Machine;
