// comfydeploy/sdk/sdk.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ComfyDeploy = void 0;
const sdks_js_1 = require("../lib/sdks.js");
const beta_js_1 = require("./beta.js");
const deployments_js_1 = require("./deployments.js");
const file_js_1 = require("./file.js");
const run_js_1 = require("./run.js");
const session_js_1 = require("./session.js");
class ComfyDeploy extends sdks_js_1.ClientSDK {
    get run() {
        var _a;
        return ((_a = this._run) !== null && _a !== void 0 ? _a : (this._run = new run_js_1.Run(this._options)));
    }
    get session() {
        var _a;
        return ((_a = this._session) !== null && _a !== void 0 ? _a : (this._session = new session_js_1.Session(this._options)));
    }
    get beta() {
        var _a;
        return ((_a = this._beta) !== null && _a !== void 0 ? _a : (this._beta = new beta_js_1.Beta(this._options)));
    }
    get deployments() {
        var _a;
        return ((_a = this._deployments) !== null && _a !== void 0 ? _a : (this._deployments = new deployments_js_1.Deployments(this._options)));
    }
    get file() {
        var _a;
        return ((_a = this._file) !== null && _a !== void 0 ? _a : (this._file = new file_js_1.FileT(this._options)));
    }
}
exports.ComfyDeploy = ComfyDeploy;
