// comfydeploy/lib/config.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SDK_METADATA = exports.ServerList = void 0;
exports.serverURLFromOptions = serverURLFromOptions;
const url_js_1 = require("./url.js");
/**
 * Contains the list of servers available to the SDK
 */
exports.ServerList = [
    /**
     * Production server
     */
    "https://api.comfydeploy.com/api",
    /**
     * Staging server
     */
    "https://staging.api.comfydeploy.com/api",
    /**
     * Local development server
     */
    "http://localhost:3011/api",
];
function serverURLFromOptions(options) {
    var _a;
    let serverURL = options.serverURL;
    const params = {};
    if (!serverURL) {
        const serverIdx = (_a = options.serverIdx) !== null && _a !== void 0 ? _a : 0;
        if (serverIdx < 0 || serverIdx >= exports.ServerList.length) {
            throw new Error(`Invalid server index ${serverIdx}`);
        }
        serverURL = exports.ServerList[serverIdx] || "";
    }
    const u = (0, url_js_1.pathToFunc)(serverURL)(params);
    return new URL(u);
}
exports.SDK_METADATA = {
    language: "typescript",
    openapiDocVersion: "1.0.0",
    sdkVersion: "2.0.0-beta.59",
    genVersion: "2.438.15",
    userAgent: "speakeasy-sdk/typescript 2.0.0-beta.59 2.438.15 1.0.0 comfydeploy",
};
