// comfydeploy/models/operations/createrunallrunpost.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateRunAllRunPostResponseCreateRunAllRunPost$ = exports.CreateRunAllRunPostResponseCreateRunAllRunPost$outboundSchema = exports.CreateRunAllRunPostResponseCreateRunAllRunPost$inboundSchema = exports.CreateRunAllRunPostData$ = exports.CreateRunAllRunPostData$outboundSchema = exports.CreateRunAllRunPostData$inboundSchema = void 0;
const z = __importStar(require("zod"));
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.CreateRunAllRunPostData$inboundSchema = z.union([
    components.DeploymentRunRequest$inboundSchema,
    components.WorkflowRunVersionRequest$inboundSchema,
    components.WorkflowRunRequest$inboundSchema,
]);
/** @internal */
exports.CreateRunAllRunPostData$outboundSchema = z.union([
    components.DeploymentRunRequest$outboundSchema,
    components.WorkflowRunVersionRequest$outboundSchema,
    components.WorkflowRunRequest$outboundSchema,
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CreateRunAllRunPostData$;
(function (CreateRunAllRunPostData$) {
    /** @deprecated use `CreateRunAllRunPostData$inboundSchema` instead. */
    CreateRunAllRunPostData$.inboundSchema = exports.CreateRunAllRunPostData$inboundSchema;
    /** @deprecated use `CreateRunAllRunPostData$outboundSchema` instead. */
    CreateRunAllRunPostData$.outboundSchema = exports.CreateRunAllRunPostData$outboundSchema;
})(CreateRunAllRunPostData$ || (exports.CreateRunAllRunPostData$ = CreateRunAllRunPostData$ = {}));
/** @internal */
exports.CreateRunAllRunPostResponseCreateRunAllRunPost$inboundSchema = z.union([
    components.CreateRunResponse$inboundSchema,
    components.CreateRunBatchResponse$inboundSchema,
    components.WorkflowRunNativeOutputModel$inboundSchema,
    components.ApiRoutesTypesWorkflowRunOutputModel2$inboundSchema,
]);
/** @internal */
exports.CreateRunAllRunPostResponseCreateRunAllRunPost$outboundSchema = z.union([
    components.CreateRunResponse$outboundSchema,
    components.CreateRunBatchResponse$outboundSchema,
    components.WorkflowRunNativeOutputModel$outboundSchema,
    components.ApiRoutesTypesWorkflowRunOutputModel2$outboundSchema,
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CreateRunAllRunPostResponseCreateRunAllRunPost$;
(function (CreateRunAllRunPostResponseCreateRunAllRunPost$) {
    /** @deprecated use `CreateRunAllRunPostResponseCreateRunAllRunPost$inboundSchema` instead. */
    CreateRunAllRunPostResponseCreateRunAllRunPost$.inboundSchema = exports.CreateRunAllRunPostResponseCreateRunAllRunPost$inboundSchema;
    /** @deprecated use `CreateRunAllRunPostResponseCreateRunAllRunPost$outboundSchema` instead. */
    CreateRunAllRunPostResponseCreateRunAllRunPost$.outboundSchema = exports.CreateRunAllRunPostResponseCreateRunAllRunPost$outboundSchema;
})(CreateRunAllRunPostResponseCreateRunAllRunPost$ || (exports.CreateRunAllRunPostResponseCreateRunAllRunPost$ = CreateRunAllRunPostResponseCreateRunAllRunPost$ = {}));
