// comfydeploy/models/operations/getauthresponserequestid.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetAuthResponseRequestIdResponse$ = exports.GetAuthResponseRequestIdResponse$outboundSchema = exports.GetAuthResponseRequestIdResponse$inboundSchema = exports.GetAuthResponseRequestIdResponseBody$ = exports.GetAuthResponseRequestIdResponseBody$outboundSchema = exports.GetAuthResponseRequestIdResponseBody$inboundSchema = exports.GetAuthResponseRequestIdComfyuiResponseBody$ = exports.GetAuthResponseRequestIdComfyuiResponseBody$outboundSchema = exports.GetAuthResponseRequestIdComfyuiResponseBody$inboundSchema = exports.GetAuthResponseRequestIdRequest$ = exports.GetAuthResponseRequestIdRequest$outboundSchema = exports.GetAuthResponseRequestIdRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
/** @internal */
exports.GetAuthResponseRequestIdRequest$inboundSchema = z.object({
    request_id: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "request_id": "requestId",
    });
});
/** @internal */
exports.GetAuthResponseRequestIdRequest$outboundSchema = z.object({
    requestId: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        requestId: "request_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetAuthResponseRequestIdRequest$;
(function (GetAuthResponseRequestIdRequest$) {
    /** @deprecated use `GetAuthResponseRequestIdRequest$inboundSchema` instead. */
    GetAuthResponseRequestIdRequest$.inboundSchema = exports.GetAuthResponseRequestIdRequest$inboundSchema;
    /** @deprecated use `GetAuthResponseRequestIdRequest$outboundSchema` instead. */
    GetAuthResponseRequestIdRequest$.outboundSchema = exports.GetAuthResponseRequestIdRequest$outboundSchema;
})(GetAuthResponseRequestIdRequest$ || (exports.GetAuthResponseRequestIdRequest$ = GetAuthResponseRequestIdRequest$ = {}));
/** @internal */
exports.GetAuthResponseRequestIdComfyuiResponseBody$inboundSchema = z.object({
    message: z.string(),
});
/** @internal */
exports.GetAuthResponseRequestIdComfyuiResponseBody$outboundSchema = z.object({
    message: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetAuthResponseRequestIdComfyuiResponseBody$;
(function (GetAuthResponseRequestIdComfyuiResponseBody$) {
    /** @deprecated use `GetAuthResponseRequestIdComfyuiResponseBody$inboundSchema` instead. */
    GetAuthResponseRequestIdComfyuiResponseBody$.inboundSchema = exports.GetAuthResponseRequestIdComfyuiResponseBody$inboundSchema;
    /** @deprecated use `GetAuthResponseRequestIdComfyuiResponseBody$outboundSchema` instead. */
    GetAuthResponseRequestIdComfyuiResponseBody$.outboundSchema = exports.GetAuthResponseRequestIdComfyuiResponseBody$outboundSchema;
})(GetAuthResponseRequestIdComfyuiResponseBody$ || (exports.GetAuthResponseRequestIdComfyuiResponseBody$ = GetAuthResponseRequestIdComfyuiResponseBody$ = {}));
/** @internal */
exports.GetAuthResponseRequestIdResponseBody$inboundSchema = z.object({
    api_key: z.string(),
    name: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "api_key": "apiKey",
    });
});
/** @internal */
exports.GetAuthResponseRequestIdResponseBody$outboundSchema = z.object({
    apiKey: z.string(),
    name: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        apiKey: "api_key",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetAuthResponseRequestIdResponseBody$;
(function (GetAuthResponseRequestIdResponseBody$) {
    /** @deprecated use `GetAuthResponseRequestIdResponseBody$inboundSchema` instead. */
    GetAuthResponseRequestIdResponseBody$.inboundSchema = exports.GetAuthResponseRequestIdResponseBody$inboundSchema;
    /** @deprecated use `GetAuthResponseRequestIdResponseBody$outboundSchema` instead. */
    GetAuthResponseRequestIdResponseBody$.outboundSchema = exports.GetAuthResponseRequestIdResponseBody$outboundSchema;
})(GetAuthResponseRequestIdResponseBody$ || (exports.GetAuthResponseRequestIdResponseBody$ = GetAuthResponseRequestIdResponseBody$ = {}));
/** @internal */
exports.GetAuthResponseRequestIdResponse$inboundSchema = z.union([
    z.lazy(() => exports.GetAuthResponseRequestIdComfyuiResponseBody$inboundSchema),
    z.lazy(() => exports.GetAuthResponseRequestIdResponseBody$inboundSchema),
]);
/** @internal */
exports.GetAuthResponseRequestIdResponse$outboundSchema = z.union([
    z.lazy(() => exports.GetAuthResponseRequestIdComfyuiResponseBody$outboundSchema),
    z.lazy(() => exports.GetAuthResponseRequestIdResponseBody$outboundSchema),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetAuthResponseRequestIdResponse$;
(function (GetAuthResponseRequestIdResponse$) {
    /** @deprecated use `GetAuthResponseRequestIdResponse$inboundSchema` instead. */
    GetAuthResponseRequestIdResponse$.inboundSchema = exports.GetAuthResponseRequestIdResponse$inboundSchema;
    /** @deprecated use `GetAuthResponseRequestIdResponse$outboundSchema` instead. */
    GetAuthResponseRequestIdResponse$.outboundSchema = exports.GetAuthResponseRequestIdResponse$outboundSchema;
})(GetAuthResponseRequestIdResponse$ || (exports.GetAuthResponseRequestIdResponse$ = GetAuthResponseRequestIdResponse$ = {}));
