// comfydeploy/models/operations/getdeployment.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetDeploymentResponseBody$ = exports.GetDeploymentResponseBody$outboundSchema = exports.GetDeploymentResponseBody$inboundSchema = exports.InputsDefinition$ = exports.InputsDefinition$outboundSchema = exports.InputsDefinition$inboundSchema = exports.GetDeploymentDefaultValue$ = exports.GetDeploymentDefaultValue$outboundSchema = exports.GetDeploymentDefaultValue$inboundSchema = exports.GetDeploymentClassType$ = exports.GetDeploymentClassType$outboundSchema = exports.GetDeploymentClassType$inboundSchema = exports.GetDeploymentRequest$ = exports.GetDeploymentRequest$outboundSchema = exports.GetDeploymentRequest$inboundSchema = exports.Environment$ = exports.Environment$outboundSchema = exports.Environment$inboundSchema = exports.GetDeploymentClassType = exports.Environment = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
exports.Environment = {
    Production: "production",
    Staging: "staging",
};
exports.GetDeploymentClassType = {
    ComfyUIDeployExternalText: "ComfyUIDeployExternalText",
    ComfyUIDeployExternalTextAny: "ComfyUIDeployExternalTextAny",
    ComfyUIDeployExternalTextSingleLine: "ComfyUIDeployExternalTextSingleLine",
    ComfyUIDeployExternalImage: "ComfyUIDeployExternalImage",
    ComfyUIDeployExternalImageAlpha: "ComfyUIDeployExternalImageAlpha",
    ComfyUIDeployExternalNumber: "ComfyUIDeployExternalNumber",
    ComfyUIDeployExternalNumberInt: "ComfyUIDeployExternalNumberInt",
    ComfyUIDeployExternalLora: "ComfyUIDeployExternalLora",
    ComfyUIDeployExternalCheckpoint: "ComfyUIDeployExternalCheckpoint",
    ComfyDeployWebscoketImageInput: "ComfyDeployWebscoketImageInput",
    ComfyUIDeployExternalImageBatch: "ComfyUIDeployExternalImageBatch",
    ComfyUIDeployExternalVideo: "ComfyUIDeployExternalVideo",
    ComfyUIDeployExternalBoolean: "ComfyUIDeployExternalBoolean",
    ComfyUIDeployExternalNumberSlider: "ComfyUIDeployExternalNumberSlider",
    ComfyUIDeployExternalNumberSliderInt: "ComfyUIDeployExternalNumberSliderInt",
    ComfyUIDeployExternalEnum: "ComfyUIDeployExternalEnum",
};
/** @internal */
exports.Environment$inboundSchema = z
    .nativeEnum(exports.Environment);
/** @internal */
exports.Environment$outboundSchema = exports.Environment$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Environment$;
(function (Environment$) {
    /** @deprecated use `Environment$inboundSchema` instead. */
    Environment$.inboundSchema = exports.Environment$inboundSchema;
    /** @deprecated use `Environment$outboundSchema` instead. */
    Environment$.outboundSchema = exports.Environment$outboundSchema;
})(Environment$ || (exports.Environment$ = Environment$ = {}));
/** @internal */
exports.GetDeploymentRequest$inboundSchema = z.object({
    environment: exports.Environment$inboundSchema.optional(),
});
/** @internal */
exports.GetDeploymentRequest$outboundSchema = z.object({
    environment: exports.Environment$outboundSchema.optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetDeploymentRequest$;
(function (GetDeploymentRequest$) {
    /** @deprecated use `GetDeploymentRequest$inboundSchema` instead. */
    GetDeploymentRequest$.inboundSchema = exports.GetDeploymentRequest$inboundSchema;
    /** @deprecated use `GetDeploymentRequest$outboundSchema` instead. */
    GetDeploymentRequest$.outboundSchema = exports.GetDeploymentRequest$outboundSchema;
})(GetDeploymentRequest$ || (exports.GetDeploymentRequest$ = GetDeploymentRequest$ = {}));
/** @internal */
exports.GetDeploymentClassType$inboundSchema = z.nativeEnum(exports.GetDeploymentClassType);
/** @internal */
exports.GetDeploymentClassType$outboundSchema = exports.GetDeploymentClassType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetDeploymentClassType$;
(function (GetDeploymentClassType$) {
    /** @deprecated use `GetDeploymentClassType$inboundSchema` instead. */
    GetDeploymentClassType$.inboundSchema = exports.GetDeploymentClassType$inboundSchema;
    /** @deprecated use `GetDeploymentClassType$outboundSchema` instead. */
    GetDeploymentClassType$.outboundSchema = exports.GetDeploymentClassType$outboundSchema;
})(GetDeploymentClassType$ || (exports.GetDeploymentClassType$ = GetDeploymentClassType$ = {}));
/** @internal */
exports.GetDeploymentDefaultValue$inboundSchema = z.union([z.string(), z.number()]);
/** @internal */
exports.GetDeploymentDefaultValue$outboundSchema = z.union([z.string(), z.number()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetDeploymentDefaultValue$;
(function (GetDeploymentDefaultValue$) {
    /** @deprecated use `GetDeploymentDefaultValue$inboundSchema` instead. */
    GetDeploymentDefaultValue$.inboundSchema = exports.GetDeploymentDefaultValue$inboundSchema;
    /** @deprecated use `GetDeploymentDefaultValue$outboundSchema` instead. */
    GetDeploymentDefaultValue$.outboundSchema = exports.GetDeploymentDefaultValue$outboundSchema;
})(GetDeploymentDefaultValue$ || (exports.GetDeploymentDefaultValue$ = GetDeploymentDefaultValue$ = {}));
/** @internal */
exports.InputsDefinition$inboundSchema = z.object({
    class_type: exports.GetDeploymentClassType$inboundSchema,
    input_id: z.string(),
    default_value: z.union([z.string(), z.number()]),
    min_value: z.number().optional(),
    max_value: z.number().optional(),
    display_name: z.string().optional(),
    description: z.string().optional(),
    enum_values: z.array(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "class_type": "classType",
        "input_id": "inputId",
        "default_value": "defaultValue",
        "min_value": "minValue",
        "max_value": "maxValue",
        "display_name": "displayName",
        "enum_values": "enumValues",
    });
});
/** @internal */
exports.InputsDefinition$outboundSchema = z.object({
    classType: exports.GetDeploymentClassType$outboundSchema,
    inputId: z.string(),
    defaultValue: z.union([z.string(), z.number()]),
    minValue: z.number().optional(),
    maxValue: z.number().optional(),
    displayName: z.string().optional(),
    description: z.string().optional(),
    enumValues: z.array(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        classType: "class_type",
        inputId: "input_id",
        defaultValue: "default_value",
        minValue: "min_value",
        maxValue: "max_value",
        displayName: "display_name",
        enumValues: "enum_values",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var InputsDefinition$;
(function (InputsDefinition$) {
    /** @deprecated use `InputsDefinition$inboundSchema` instead. */
    InputsDefinition$.inboundSchema = exports.InputsDefinition$inboundSchema;
    /** @deprecated use `InputsDefinition$outboundSchema` instead. */
    InputsDefinition$.outboundSchema = exports.InputsDefinition$outboundSchema;
})(InputsDefinition$ || (exports.InputsDefinition$ = InputsDefinition$ = {}));
/** @internal */
exports.GetDeploymentResponseBody$inboundSchema = z.object({
    deployment_id: z.string(),
    name: z.string(),
    inputs_definition: z.nullable(z.array(z.lazy(() => exports.InputsDefinition$inboundSchema))),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "deployment_id": "deploymentId",
        "inputs_definition": "inputsDefinition",
    });
});
/** @internal */
exports.GetDeploymentResponseBody$outboundSchema = z.object({
    deploymentId: z.string(),
    name: z.string(),
    inputsDefinition: z.nullable(z.array(z.lazy(() => exports.InputsDefinition$outboundSchema))),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        deploymentId: "deployment_id",
        inputsDefinition: "inputs_definition",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetDeploymentResponseBody$;
(function (GetDeploymentResponseBody$) {
    /** @deprecated use `GetDeploymentResponseBody$inboundSchema` instead. */
    GetDeploymentResponseBody$.inboundSchema = exports.GetDeploymentResponseBody$inboundSchema;
    /** @deprecated use `GetDeploymentResponseBody$outboundSchema` instead. */
    GetDeploymentResponseBody$.outboundSchema = exports.GetDeploymentResponseBody$outboundSchema;
})(GetDeploymentResponseBody$ || (exports.GetDeploymentResponseBody$ = GetDeploymentResponseBody$ = {}));
