// comfydeploy/models/operations/getdeploymentidinputs.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResponseBody$ = exports.ResponseBody$outboundSchema = exports.ResponseBody$inboundSchema = exports.DefaultValue$ = exports.DefaultValue$outboundSchema = exports.DefaultValue$inboundSchema = exports.ClassType$ = exports.ClassType$outboundSchema = exports.ClassType$inboundSchema = exports.GetDeploymentIdInputsRequest$ = exports.GetDeploymentIdInputsRequest$outboundSchema = exports.GetDeploymentIdInputsRequest$inboundSchema = exports.ClassType = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
exports.ClassType = {
    ComfyUIDeployExternalText: "ComfyUIDeployExternalText",
    ComfyUIDeployExternalTextAny: "ComfyUIDeployExternalTextAny",
    ComfyUIDeployExternalTextSingleLine: "ComfyUIDeployExternalTextSingleLine",
    ComfyUIDeployExternalImage: "ComfyUIDeployExternalImage",
    ComfyUIDeployExternalImageAlpha: "ComfyUIDeployExternalImageAlpha",
    ComfyUIDeployExternalNumber: "ComfyUIDeployExternalNumber",
    ComfyUIDeployExternalNumberInt: "ComfyUIDeployExternalNumberInt",
    ComfyUIDeployExternalLora: "ComfyUIDeployExternalLora",
    ComfyUIDeployExternalCheckpoint: "ComfyUIDeployExternalCheckpoint",
    ComfyDeployWebscoketImageInput: "ComfyDeployWebscoketImageInput",
    ComfyUIDeployExternalImageBatch: "ComfyUIDeployExternalImageBatch",
    ComfyUIDeployExternalVideo: "ComfyUIDeployExternalVideo",
    ComfyUIDeployExternalBoolean: "ComfyUIDeployExternalBoolean",
    ComfyUIDeployExternalNumberSlider: "ComfyUIDeployExternalNumberSlider",
    ComfyUIDeployExternalNumberSliderInt: "ComfyUIDeployExternalNumberSliderInt",
    ComfyUIDeployExternalEnum: "ComfyUIDeployExternalEnum",
};
/** @internal */
exports.GetDeploymentIdInputsRequest$inboundSchema = z.object({
    id: z.string(),
});
/** @internal */
exports.GetDeploymentIdInputsRequest$outboundSchema = z.object({
    id: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetDeploymentIdInputsRequest$;
(function (GetDeploymentIdInputsRequest$) {
    /** @deprecated use `GetDeploymentIdInputsRequest$inboundSchema` instead. */
    GetDeploymentIdInputsRequest$.inboundSchema = exports.GetDeploymentIdInputsRequest$inboundSchema;
    /** @deprecated use `GetDeploymentIdInputsRequest$outboundSchema` instead. */
    GetDeploymentIdInputsRequest$.outboundSchema = exports.GetDeploymentIdInputsRequest$outboundSchema;
})(GetDeploymentIdInputsRequest$ || (exports.GetDeploymentIdInputsRequest$ = GetDeploymentIdInputsRequest$ = {}));
/** @internal */
exports.ClassType$inboundSchema = z
    .nativeEnum(exports.ClassType);
/** @internal */
exports.ClassType$outboundSchema = exports.ClassType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ClassType$;
(function (ClassType$) {
    /** @deprecated use `ClassType$inboundSchema` instead. */
    ClassType$.inboundSchema = exports.ClassType$inboundSchema;
    /** @deprecated use `ClassType$outboundSchema` instead. */
    ClassType$.outboundSchema = exports.ClassType$outboundSchema;
})(ClassType$ || (exports.ClassType$ = ClassType$ = {}));
/** @internal */
exports.DefaultValue$inboundSchema = z.union([z.string(), z.number()]);
/** @internal */
exports.DefaultValue$outboundSchema = z.union([z.string(), z.number()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DefaultValue$;
(function (DefaultValue$) {
    /** @deprecated use `DefaultValue$inboundSchema` instead. */
    DefaultValue$.inboundSchema = exports.DefaultValue$inboundSchema;
    /** @deprecated use `DefaultValue$outboundSchema` instead. */
    DefaultValue$.outboundSchema = exports.DefaultValue$outboundSchema;
})(DefaultValue$ || (exports.DefaultValue$ = DefaultValue$ = {}));
/** @internal */
exports.ResponseBody$inboundSchema = z.object({
    class_type: exports.ClassType$inboundSchema,
    input_id: z.string(),
    default_value: z.union([z.string(), z.number()]),
    min_value: z.number().optional(),
    max_value: z.number().optional(),
    display_name: z.string().optional(),
    description: z.string().optional(),
    enum_values: z.array(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "class_type": "classType",
        "input_id": "inputId",
        "default_value": "defaultValue",
        "min_value": "minValue",
        "max_value": "maxValue",
        "display_name": "displayName",
        "enum_values": "enumValues",
    });
});
/** @internal */
exports.ResponseBody$outboundSchema = z.object({
    classType: exports.ClassType$outboundSchema,
    inputId: z.string(),
    defaultValue: z.union([z.string(), z.number()]),
    minValue: z.number().optional(),
    maxValue: z.number().optional(),
    displayName: z.string().optional(),
    description: z.string().optional(),
    enumValues: z.array(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        classType: "class_type",
        inputId: "input_id",
        defaultValue: "default_value",
        minValue: "min_value",
        maxValue: "max_value",
        displayName: "display_name",
        enumValues: "enum_values",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ResponseBody$;
(function (ResponseBody$) {
    /** @deprecated use `ResponseBody$inboundSchema` instead. */
    ResponseBody$.inboundSchema = exports.ResponseBody$inboundSchema;
    /** @deprecated use `ResponseBody$outboundSchema` instead. */
    ResponseBody$.outboundSchema = exports.ResponseBody$outboundSchema;
})(ResponseBody$ || (exports.ResponseBody$ = ResponseBody$ = {}));
