// comfydeploy/models/operations/getmachinemachineidevents.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetMachineMachineIdEventsResponseBody$ = exports.GetMachineMachineIdEventsResponseBody$outboundSchema = exports.GetMachineMachineIdEventsResponseBody$inboundSchema = exports.ProviderType$ = exports.ProviderType$outboundSchema = exports.ProviderType$inboundSchema = exports.WsGpu$ = exports.WsGpu$outboundSchema = exports.WsGpu$inboundSchema = exports.GetMachineMachineIdEventsGpu$ = exports.GetMachineMachineIdEventsGpu$outboundSchema = exports.GetMachineMachineIdEventsGpu$inboundSchema = exports.GetMachineMachineIdEventsRequest$ = exports.GetMachineMachineIdEventsRequest$outboundSchema = exports.GetMachineMachineIdEventsRequest$inboundSchema = exports.Status$ = exports.Status$outboundSchema = exports.Status$inboundSchema = exports.ProviderType = exports.WsGpu = exports.GetMachineMachineIdEventsGpu = exports.Status = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
exports.Status = {
    Running: "running",
    Completed: "completed",
};
exports.GetMachineMachineIdEventsGpu = {
    T4: "T4",
    L4: "L4",
    A10G: "A10G",
    A100: "A100",
    A10080Gb: "A100-80GB",
    H100: "H100",
};
exports.WsGpu = {
    FourThousandAndNinety: "4090",
};
exports.ProviderType = {
    Modal: "modal",
    Runpod: "runpod",
};
/** @internal */
exports.Status$inboundSchema = z
    .nativeEnum(exports.Status);
/** @internal */
exports.Status$outboundSchema = exports.Status$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Status$;
(function (Status$) {
    /** @deprecated use `Status$inboundSchema` instead. */
    Status$.inboundSchema = exports.Status$inboundSchema;
    /** @deprecated use `Status$outboundSchema` instead. */
    Status$.outboundSchema = exports.Status$outboundSchema;
})(Status$ || (exports.Status$ = Status$ = {}));
/** @internal */
exports.GetMachineMachineIdEventsRequest$inboundSchema = z.object({
    machine_id: z.string(),
    status: exports.Status$inboundSchema,
    limit: z.number().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "machine_id": "machineId",
    });
});
/** @internal */
exports.GetMachineMachineIdEventsRequest$outboundSchema = z.object({
    machineId: z.string(),
    status: exports.Status$outboundSchema,
    limit: z.number().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        machineId: "machine_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetMachineMachineIdEventsRequest$;
(function (GetMachineMachineIdEventsRequest$) {
    /** @deprecated use `GetMachineMachineIdEventsRequest$inboundSchema` instead. */
    GetMachineMachineIdEventsRequest$.inboundSchema = exports.GetMachineMachineIdEventsRequest$inboundSchema;
    /** @deprecated use `GetMachineMachineIdEventsRequest$outboundSchema` instead. */
    GetMachineMachineIdEventsRequest$.outboundSchema = exports.GetMachineMachineIdEventsRequest$outboundSchema;
})(GetMachineMachineIdEventsRequest$ || (exports.GetMachineMachineIdEventsRequest$ = GetMachineMachineIdEventsRequest$ = {}));
/** @internal */
exports.GetMachineMachineIdEventsGpu$inboundSchema = z.nativeEnum(exports.GetMachineMachineIdEventsGpu);
/** @internal */
exports.GetMachineMachineIdEventsGpu$outboundSchema = exports.GetMachineMachineIdEventsGpu$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetMachineMachineIdEventsGpu$;
(function (GetMachineMachineIdEventsGpu$) {
    /** @deprecated use `GetMachineMachineIdEventsGpu$inboundSchema` instead. */
    GetMachineMachineIdEventsGpu$.inboundSchema = exports.GetMachineMachineIdEventsGpu$inboundSchema;
    /** @deprecated use `GetMachineMachineIdEventsGpu$outboundSchema` instead. */
    GetMachineMachineIdEventsGpu$.outboundSchema = exports.GetMachineMachineIdEventsGpu$outboundSchema;
})(GetMachineMachineIdEventsGpu$ || (exports.GetMachineMachineIdEventsGpu$ = GetMachineMachineIdEventsGpu$ = {}));
/** @internal */
exports.WsGpu$inboundSchema = z.nativeEnum(exports.WsGpu);
/** @internal */
exports.WsGpu$outboundSchema = exports.WsGpu$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WsGpu$;
(function (WsGpu$) {
    /** @deprecated use `WsGpu$inboundSchema` instead. */
    WsGpu$.inboundSchema = exports.WsGpu$inboundSchema;
    /** @deprecated use `WsGpu$outboundSchema` instead. */
    WsGpu$.outboundSchema = exports.WsGpu$outboundSchema;
})(WsGpu$ || (exports.WsGpu$ = WsGpu$ = {}));
/** @internal */
exports.ProviderType$inboundSchema = z.nativeEnum(exports.ProviderType);
/** @internal */
exports.ProviderType$outboundSchema = exports.ProviderType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProviderType$;
(function (ProviderType$) {
    /** @deprecated use `ProviderType$inboundSchema` instead. */
    ProviderType$.inboundSchema = exports.ProviderType$inboundSchema;
    /** @deprecated use `ProviderType$outboundSchema` instead. */
    ProviderType$.outboundSchema = exports.ProviderType$outboundSchema;
})(ProviderType$ || (exports.ProviderType$ = ProviderType$ = {}));
/** @internal */
exports.GetMachineMachineIdEventsResponseBody$inboundSchema = z.object({
    id: z.string(),
    user_id: z.string(),
    org_id: z.nullable(z.string()),
    machine_id: z.nullable(z.string()),
    start_time: z.nullable(z.string()),
    end_time: z.nullable(z.string()),
    gpu: z.nullable(exports.GetMachineMachineIdEventsGpu$inboundSchema),
    ws_gpu: z.nullable(exports.WsGpu$inboundSchema),
    providerType: exports.ProviderType$inboundSchema,
    created_at: z.string(),
    updated_at: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "user_id": "userId",
        "org_id": "orgId",
        "machine_id": "machineId",
        "start_time": "startTime",
        "end_time": "endTime",
        "ws_gpu": "wsGpu",
        "created_at": "createdAt",
        "updated_at": "updatedAt",
    });
});
/** @internal */
exports.GetMachineMachineIdEventsResponseBody$outboundSchema = z.object({
    id: z.string(),
    userId: z.string(),
    orgId: z.nullable(z.string()),
    machineId: z.nullable(z.string()),
    startTime: z.nullable(z.string()),
    endTime: z.nullable(z.string()),
    gpu: z.nullable(exports.GetMachineMachineIdEventsGpu$outboundSchema),
    wsGpu: z.nullable(exports.WsGpu$outboundSchema),
    providerType: exports.ProviderType$outboundSchema,
    createdAt: z.string(),
    updatedAt: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        userId: "user_id",
        orgId: "org_id",
        machineId: "machine_id",
        startTime: "start_time",
        endTime: "end_time",
        wsGpu: "ws_gpu",
        createdAt: "created_at",
        updatedAt: "updated_at",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetMachineMachineIdEventsResponseBody$;
(function (GetMachineMachineIdEventsResponseBody$) {
    /** @deprecated use `GetMachineMachineIdEventsResponseBody$inboundSchema` instead. */
    GetMachineMachineIdEventsResponseBody$.inboundSchema = exports.GetMachineMachineIdEventsResponseBody$inboundSchema;
    /** @deprecated use `GetMachineMachineIdEventsResponseBody$outboundSchema` instead. */
    GetMachineMachineIdEventsResponseBody$.outboundSchema = exports.GetMachineMachineIdEventsResponseBody$outboundSchema;
})(GetMachineMachineIdEventsResponseBody$ || (exports.GetMachineMachineIdEventsResponseBody$ = GetMachineMachineIdEventsResponseBody$ = {}));
