// comfydeploy/models/operations/getv1machinesmachineid.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetV1MachinesMachineIdResponseBody$ = exports.GetV1MachinesMachineIdResponseBody$outboundSchema = exports.GetV1MachinesMachineIdResponseBody$inboundSchema = exports.GetV1MachinesMachineIdRequest$ = exports.GetV1MachinesMachineIdRequest$outboundSchema = exports.GetV1MachinesMachineIdRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
/** @internal */
exports.GetV1MachinesMachineIdRequest$inboundSchema = z.object({
    machine_id: z.string(),
    extUrls: z.string().default("false"),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "machine_id": "machineId",
    });
});
/** @internal */
exports.GetV1MachinesMachineIdRequest$outboundSchema = z.object({
    machineId: z.string(),
    extUrls: z.string().default("false"),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        machineId: "machine_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetV1MachinesMachineIdRequest$;
(function (GetV1MachinesMachineIdRequest$) {
    /** @deprecated use `GetV1MachinesMachineIdRequest$inboundSchema` instead. */
    GetV1MachinesMachineIdRequest$.inboundSchema = exports.GetV1MachinesMachineIdRequest$inboundSchema;
    /** @deprecated use `GetV1MachinesMachineIdRequest$outboundSchema` instead. */
    GetV1MachinesMachineIdRequest$.outboundSchema = exports.GetV1MachinesMachineIdRequest$outboundSchema;
})(GetV1MachinesMachineIdRequest$ || (exports.GetV1MachinesMachineIdRequest$ = GetV1MachinesMachineIdRequest$ = {}));
/** @internal */
exports.GetV1MachinesMachineIdResponseBody$inboundSchema = z.object({
    id: z.string().optional(),
    name: z.string().optional(),
    type: z.string().optional(),
    status: z.string().optional(),
    created_at: z.string().optional(),
    updated_at: z.string().optional(),
    message: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "updated_at": "updatedAt",
    });
});
/** @internal */
exports.GetV1MachinesMachineIdResponseBody$outboundSchema = z.object({
    id: z.string().optional(),
    name: z.string().optional(),
    type: z.string().optional(),
    status: z.string().optional(),
    createdAt: z.string().optional(),
    updatedAt: z.string().optional(),
    message: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        updatedAt: "updated_at",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetV1MachinesMachineIdResponseBody$;
(function (GetV1MachinesMachineIdResponseBody$) {
    /** @deprecated use `GetV1MachinesMachineIdResponseBody$inboundSchema` instead. */
    GetV1MachinesMachineIdResponseBody$.inboundSchema = exports.GetV1MachinesMachineIdResponseBody$inboundSchema;
    /** @deprecated use `GetV1MachinesMachineIdResponseBody$outboundSchema` instead. */
    GetV1MachinesMachineIdResponseBody$.outboundSchema = exports.GetV1MachinesMachineIdResponseBody$outboundSchema;
})(GetV1MachinesMachineIdResponseBody$ || (exports.GetV1MachinesMachineIdResponseBody$ = GetV1MachinesMachineIdResponseBody$ = {}));
