// comfydeploy/models/operations/getworkflowversionversionid.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetWorkflowVersionVersionIdResponseBody$ = exports.GetWorkflowVersionVersionIdResponseBody$outboundSchema = exports.GetWorkflowVersionVersionIdResponseBody$inboundSchema = exports.GetWorkflowVersionVersionIdRequest$ = exports.GetWorkflowVersionVersionIdRequest$outboundSchema = exports.GetWorkflowVersionVersionIdRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
/** @internal */
exports.GetWorkflowVersionVersionIdRequest$inboundSchema = z.object({
    version_id: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "version_id": "versionId",
    });
});
/** @internal */
exports.GetWorkflowVersionVersionIdRequest$outboundSchema = z.object({
    versionId: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        versionId: "version_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetWorkflowVersionVersionIdRequest$;
(function (GetWorkflowVersionVersionIdRequest$) {
    /** @deprecated use `GetWorkflowVersionVersionIdRequest$inboundSchema` instead. */
    GetWorkflowVersionVersionIdRequest$.inboundSchema = exports.GetWorkflowVersionVersionIdRequest$inboundSchema;
    /** @deprecated use `GetWorkflowVersionVersionIdRequest$outboundSchema` instead. */
    GetWorkflowVersionVersionIdRequest$.outboundSchema = exports.GetWorkflowVersionVersionIdRequest$outboundSchema;
})(GetWorkflowVersionVersionIdRequest$ || (exports.GetWorkflowVersionVersionIdRequest$ = GetWorkflowVersionVersionIdRequest$ = {}));
/** @internal */
exports.GetWorkflowVersionVersionIdResponseBody$inboundSchema = z.object({
    workflow_id: z.string(),
    id: z.string(),
    workflow: z.any().optional(),
    workflow_api: z.any().optional(),
    user_id: z.nullable(z.string()),
    comment: z.nullable(z.string()),
    version: z.number(),
    snapshot: z.any().optional(),
    dependencies: z.any().optional(),
    created_at: z.string(),
    updated_at: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "workflow_id": "workflowId",
        "workflow_api": "workflowApi",
        "user_id": "userId",
        "created_at": "createdAt",
        "updated_at": "updatedAt",
    });
});
/** @internal */
exports.GetWorkflowVersionVersionIdResponseBody$outboundSchema = z.object({
    workflowId: z.string(),
    id: z.string(),
    workflow: z.any().optional(),
    workflowApi: z.any().optional(),
    userId: z.nullable(z.string()),
    comment: z.nullable(z.string()),
    version: z.number(),
    snapshot: z.any().optional(),
    dependencies: z.any().optional(),
    createdAt: z.string(),
    updatedAt: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        workflowId: "workflow_id",
        workflowApi: "workflow_api",
        userId: "user_id",
        createdAt: "created_at",
        updatedAt: "updated_at",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetWorkflowVersionVersionIdResponseBody$;
(function (GetWorkflowVersionVersionIdResponseBody$) {
    /** @deprecated use `GetWorkflowVersionVersionIdResponseBody$inboundSchema` instead. */
    GetWorkflowVersionVersionIdResponseBody$.inboundSchema = exports.GetWorkflowVersionVersionIdResponseBody$inboundSchema;
    /** @deprecated use `GetWorkflowVersionVersionIdResponseBody$outboundSchema` instead. */
    GetWorkflowVersionVersionIdResponseBody$.outboundSchema = exports.GetWorkflowVersionVersionIdResponseBody$outboundSchema;
})(GetWorkflowVersionVersionIdResponseBody$ || (exports.GetWorkflowVersionVersionIdResponseBody$ = GetWorkflowVersionVersionIdResponseBody$ = {}));
