// comfydeploy/models/operations/getwebsocketdeploymentid.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetWebsocketDeploymentIdResponseBody$ = exports.GetWebsocketDeploymentIdResponseBody$outboundSchema = exports.GetWebsocketDeploymentIdResponseBody$inboundSchema = exports.GetWebsocketDeploymentIdRequest$ = exports.GetWebsocketDeploymentIdRequest$outboundSchema = exports.GetWebsocketDeploymentIdRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
/** @internal */
exports.GetWebsocketDeploymentIdRequest$inboundSchema = z.object({
    deployment_id: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "deployment_id": "deploymentId",
    });
});
/** @internal */
exports.GetWebsocketDeploymentIdRequest$outboundSchema = z.object({
    deploymentId: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        deploymentId: "deployment_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetWebsocketDeploymentIdRequest$;
(function (GetWebsocketDeploymentIdRequest$) {
    /** @deprecated use `GetWebsocketDeploymentIdRequest$inboundSchema` instead. */
    GetWebsocketDeploymentIdRequest$.inboundSchema = exports.GetWebsocketDeploymentIdRequest$inboundSchema;
    /** @deprecated use `GetWebsocketDeploymentIdRequest$outboundSchema` instead. */
    GetWebsocketDeploymentIdRequest$.outboundSchema = exports.GetWebsocketDeploymentIdRequest$outboundSchema;
})(GetWebsocketDeploymentIdRequest$ || (exports.GetWebsocketDeploymentIdRequest$ = GetWebsocketDeploymentIdRequest$ = {}));
/** @internal */
exports.GetWebsocketDeploymentIdResponseBody$inboundSchema = z.object({
    ws_connection_url: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "ws_connection_url": "wsConnectionUrl",
    });
});
/** @internal */
exports.GetWebsocketDeploymentIdResponseBody$outboundSchema = z.object({
    wsConnectionUrl: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        wsConnectionUrl: "ws_connection_url",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetWebsocketDeploymentIdResponseBody$;
(function (GetWebsocketDeploymentIdResponseBody$) {
    /** @deprecated use `GetWebsocketDeploymentIdResponseBody$inboundSchema` instead. */
    GetWebsocketDeploymentIdResponseBody$.inboundSchema = exports.GetWebsocketDeploymentIdResponseBody$inboundSchema;
    /** @deprecated use `GetWebsocketDeploymentIdResponseBody$outboundSchema` instead. */
    GetWebsocketDeploymentIdResponseBody$.outboundSchema = exports.GetWebsocketDeploymentIdResponseBody$outboundSchema;
})(GetWebsocketDeploymentIdResponseBody$ || (exports.GetWebsocketDeploymentIdResponseBody$ = GetWebsocketDeploymentIdResponseBody$ = {}));
