// comfydeploy/models/operations/postv1workflows.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PostV1WorkflowsResponseBody$ = exports.PostV1WorkflowsResponseBody$outboundSchema = exports.PostV1WorkflowsResponseBody$inboundSchema = exports.PostV1WorkflowsRequestBody$ = exports.PostV1WorkflowsRequestBody$outboundSchema = exports.PostV1WorkflowsRequestBody$inboundSchema = exports.WorkflowJson$ = exports.WorkflowJson$outboundSchema = exports.WorkflowJson$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
/** @internal */
exports.WorkflowJson$inboundSchema = z.object({});
/** @internal */
exports.WorkflowJson$outboundSchema = z.object({});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WorkflowJson$;
(function (WorkflowJson$) {
    /** @deprecated use `WorkflowJson$inboundSchema` instead. */
    WorkflowJson$.inboundSchema = exports.WorkflowJson$inboundSchema;
    /** @deprecated use `WorkflowJson$outboundSchema` instead. */
    WorkflowJson$.outboundSchema = exports.WorkflowJson$outboundSchema;
})(WorkflowJson$ || (exports.WorkflowJson$ = WorkflowJson$ = {}));
/** @internal */
exports.PostV1WorkflowsRequestBody$inboundSchema = z.object({
    workflow_name: z.string().default("new workflow"),
    workflow_json: z.lazy(() => exports.WorkflowJson$inboundSchema).optional(),
    create_machine: z.boolean().default(true),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "workflow_name": "workflowName",
        "workflow_json": "workflowJson",
        "create_machine": "createMachine",
    });
});
/** @internal */
exports.PostV1WorkflowsRequestBody$outboundSchema = z.object({
    workflowName: z.string().default("new workflow"),
    workflowJson: z.lazy(() => exports.WorkflowJson$outboundSchema).optional(),
    createMachine: z.boolean().default(true),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        workflowName: "workflow_name",
        workflowJson: "workflow_json",
        createMachine: "create_machine",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var PostV1WorkflowsRequestBody$;
(function (PostV1WorkflowsRequestBody$) {
    /** @deprecated use `PostV1WorkflowsRequestBody$inboundSchema` instead. */
    PostV1WorkflowsRequestBody$.inboundSchema = exports.PostV1WorkflowsRequestBody$inboundSchema;
    /** @deprecated use `PostV1WorkflowsRequestBody$outboundSchema` instead. */
    PostV1WorkflowsRequestBody$.outboundSchema = exports.PostV1WorkflowsRequestBody$outboundSchema;
})(PostV1WorkflowsRequestBody$ || (exports.PostV1WorkflowsRequestBody$ = PostV1WorkflowsRequestBody$ = {}));
/** @internal */
exports.PostV1WorkflowsResponseBody$inboundSchema = z.object({
    workflow_id: z.string(),
    workflow_version: z.number(),
    machine_id: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "workflow_id": "workflowId",
        "workflow_version": "workflowVersion",
        "machine_id": "machineId",
    });
});
/** @internal */
exports.PostV1WorkflowsResponseBody$outboundSchema = z.object({
    workflowId: z.string(),
    workflowVersion: z.number(),
    machineId: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        workflowId: "workflow_id",
        workflowVersion: "workflow_version",
        machineId: "machine_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var PostV1WorkflowsResponseBody$;
(function (PostV1WorkflowsResponseBody$) {
    /** @deprecated use `PostV1WorkflowsResponseBody$inboundSchema` instead. */
    PostV1WorkflowsResponseBody$.inboundSchema = exports.PostV1WorkflowsResponseBody$inboundSchema;
    /** @deprecated use `PostV1WorkflowsResponseBody$outboundSchema` instead. */
    PostV1WorkflowsResponseBody$.outboundSchema = exports.PostV1WorkflowsResponseBody$outboundSchema;
})(PostV1WorkflowsResponseBody$ || (exports.PostV1WorkflowsResponseBody$ = PostV1WorkflowsResponseBody$ = {}));
